app/models/discovery_rule.rb

```diff
@@ -26,8 +25,16 @@ class DiscoveryRule < ActiveRecord::Base
   scoped_search :on => :enabled
   scoped_search :in => :hostgroup, :on => :name, :complete_value => true, :rename => :hostgroup
 
+  # with proc support, default_scope can no longer be chained
+  # include all default scoping here
+  default_scope lambda {
+                  with_taxonomy_scope do
+                    order("discovery_rules.name")
+                  end
+                }
+
   def default_int_attributes
-   self.max_count ||= 0
-   self.priority  ||= 0
+    self.max_count ||= 0
+    self.priority  ||= 0
```

Comment:
If we are going to cherry pick this will likely cause conflicts, but I am fine with nicer code :-)